packages/hierarchies-react/src/test/internal/TreeModel.test.ts

```diff
@@ -0,0 +1,795 @@
+import { addNodesToModel, createTestHierarchyNode, createTestModelInfoNode, createTreeModel, getHierarchyNode } from "../TestUtils";
+
+describe("TreeModel", () => {
+  describe("expandNode", () => {
+    it("expands node", () => {
+      const model = createTreeModel([
+        {
+          id: undefined,
+          children: ["root-1", "root-2"],
+        },
+        {
+          id: "root-1",
+          isExpanded: false,
+        },
+        {
+          id: "root-2",
+          isExpanded: false,
+        },
+      ]);
+
+      expect(getHierarchyNode(model, "root-1")!.isExpanded).to.be.false;
+      expect(getHierarchyNode(model, "root-2")!.isExpanded).to.be.false;
+
+      TreeModel.expandNode(model, "root-2", true);
+
+      expect(getHierarchyNode(model, "root-1")!.isExpanded).to.be.false;
+      expect(getHierarchyNode(model, "root-2")!.isExpanded).to.be.true;
+    });
+
+    it("collapses node", () => {
+      const model = createTreeModel([
+        {
+          id: undefined,
+          children: ["root-1", "root-2"],
+        },
+        {
+          id: "root-1",
+          isExpanded: true,
+        },
+        {
+          id: "root-2",
+          isExpanded: true,
+        },
+      ]);
+
+      expect(getHierarchyNode(model, "root-1")!.isExpanded).to.be.true;
+      expect(getHierarchyNode(model, "root-2")!.isExpanded).to.be.true;
+
+      TreeModel.expandNode(model, "root-2", false);
+
+      expect(getHierarchyNode(model, "root-1")!.isExpanded).to.be.true;
+      expect(getHierarchyNode(model, "root-2")!.isExpanded).to.be.false;
+    });
+
+    it("returns `loadChildren` if expanded node has unloaded children", () => {
+      const model = createTreeModel([
+        {
+          id: undefined,
+          children: ["root-1", "root-2"],
+        },
+        {
+          id: "root-1",
+          isExpanded: false,
+          children: undefined,
+        },
+      ]);
+
+      expect(TreeModel.expandNode(model, "root-1", true)).to.be.eq("loadChildren");
+    });
+
+    it("returns `none` if expanded node has loaded children", () => {
+      const model = createTreeModel([
+        {
+          id: undefined,
+          children: ["root-1", "root-2"],
+        },
+        {
+          id: "root-1",
+          isExpanded: false,
+          children: ["child-1", "child-2"],
+        },
+      ]);
+
+      expect(TreeModel.expandNode(model, "root-1", true)).to.be.eq("none");
+    });
+
+    it("returns `none` if expanded node has no children", () => {
+      const model = createTreeModel([
+        {
+          id: undefined,
+          children: ["root-1", "root-2"],
+        },
+        {
+          id: "root-1",
+          isExpanded: false,
+          children: [],
+        },
+      ]);
+
+      expect(TreeModel.expandNode(model, "root-1", true)).to.be.eq("none");
+    });
+
+    it("sets `isLoading` = `true` if expanded node has unloaded children", () => {
```

Comment:
I think this can be merged with `returns \`loadChildren\` if expanded node has unloaded children`